src/Resolver.php

```diff
@@ -524,9 +524,9 @@ public function resolveRelations($path, Model $subject = null)
         $relations = explode('.', $path);
         $subject = $subject ?: $this->query->getModel();
 
-        if ($relations[0] !== $subject->getTableName()) {
+        if ($relations[0] !== $subject->getTableAlias()) {
             throw new InvalidArgumentException(sprintf(
-                'Cannot resolve relation path "%s". Base table name is missing.',
+                'Cannot resolve relation path "%s". Base alias/table name is missing.',
```

Comment:
Nah, it's the name or alias of the base table, not the base name or base alias. My suggestion was correct already.